src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,120 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    private Node<K, V> nullKeyNode;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            if (nullKeyNode == null) {
+                nullKeyNode = new Node<>(null, value);
+                size++;
+            } else {
+                nullKeyNode.value = value;
+            }
+            return;
+        }
+
+        int index = getIndex(key, table.length);
+        Node<K, V> current = table[index];
+
+        while (current != null) {
+            if (current.key.equals(key)) {
+                current.value = value;
+                return;
+            }
+            current = current.next;
+        }
 
+        Node<K, V> newNode = new Node<>(key, value);
+        newNode.next = table[index];
+        table[index] = newNode;
+        size++;
+
+        if (size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            return nullKeyNode == null ? null : nullKeyNode.value;
+        }
+
+        int index = getIndex(key, table.length);
+        Node<K, V> current = table[index];
+
+        while (current != null) {
+            if (current.key.equals(key)) {
+                return current.value;
+            }
+            current = current.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        int newCapacity = table.length * 2;
+        Node<K, V>[] newTable = new Node[newCapacity];
+
+        for (Node<K, V> headNode : table) {
+            while (headNode != null) {
+                Node<K, V> next = headNode.next;
+                int newIndex = getIndex(headNode.key, newCapacity);
+
+                headNode.next = newTable[newIndex];
+                newTable[newIndex] = headNode;
+
+                headNode = next;
+            }
+        }
+
+        table = newTable;
+        threshold = (int) (newCapacity * LOAD_FACTOR);
+    }
+
+    private int getIndex(K key, int length) {
+        return Math.abs(key.hashCode()) % length;
```

Comment:
In your solution, if `key == null ` you will get `NullPointerException`
```suggestion
        return (key == null) ? 0 : Math.abs(key.hashCode()) % length;
```